cpp/ql/src/semmle/code/cpp/exprs/Expr.qll

```diff
@@ -1268,3 +1268,32 @@ class SpaceshipExpr extends BinaryOperation, @spaceshipexpr {
 
   override string getOperator() { result = "<=>" }
 }
+
+/**
+ * A C/C++ co_await expression
+ * ```
+ * co_await foo();
+ * ```
+ */
+class CoAwaitExpr extends UnaryOperation, @co_await {
+  override string getAPrimaryQlClass() { result = "CoAwaitExpr" }
+
+  override string getOperator() { result = "co_await" }
+
+  override int getPrecedence() { result = 16 }
+}
+
+
+/**
+ * A C/C++ co_yield expression
+ * ```
+ * co_yield 1;
+ * ```
+ */
+class CoYieldExpr extends UnaryOperation, @co_yield {
+  override string getAPrimaryQlClass() { result = "CoYieldExpr" }
+
+  override string getOperator() { result = "co_yield" }
+
+  override int getPrecedence() { result = 2 }
```

Comment:
Where do you find these precedence numbers?

Reply:
The spec doesn't explicitly give the precedence; it comes out of the grammar. You can also compare with https://en.cppreference.com/w/cpp/language/operator_precedence where we put e.g. `AddressOfExpr` and `CoAwaitExpr` in the same bucket, and `Assignment` and `CoYieldExpr` in the same bucket, although we number the buckets differently.

Reply:
Thanks!